src/main/java/io/strimzi/kafka/bridge/http/HttpSinkBridgeEndpoint.java

```diff
@@ -0,0 +1,174 @@
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.strimzi.kafka.bridge.http;
+
+import io.strimzi.kafka.bridge.Endpoint;
+import io.strimzi.kafka.bridge.SinkBridgeEndpoint;
+import io.strimzi.kafka.bridge.converter.MessageConverter;
+import io.strimzi.kafka.bridge.http.converter.HttpJsonMessageConverter;
+import io.vertx.core.Handler;
+import io.vertx.core.Vertx;
+import io.vertx.core.buffer.Buffer;
+import io.vertx.core.http.HttpServerRequest;
+import io.vertx.core.http.HttpServerResponse;
+import io.vertx.core.json.JsonObject;
+
+import java.util.UUID;
+
+public class HttpSinkBridgeEndpoint<V, K> extends SinkBridgeEndpoint<V, K> {
+
+    private HttpServerRequest httpServerRequest;
+
+    private String consumerName;
+
+    private Handler<String> consumerIdHandler;
+
+    private MessageConverter messageConverter;
+
+    HttpSinkBridgeEndpoint(Vertx vertx, HttpBridgeConfigProperties httpBridgeConfigProperties) {
+        super(vertx, httpBridgeConfigProperties);
+    }
+
+    @Override
+    public void open() {
+
+    }
+
+    @Override
+    public void handle(Endpoint<?> endpoint) {
+
+        httpServerRequest = (HttpServerRequest) endpoint.get();
+
+        messageConverter = new HttpJsonMessageConverter();
+
+        RequestType requestType = RequestIdentifier.getRequestType(httpServerRequest);
+
+        switch (requestType){
+
+            case SUBSCRIBE:
+
+                httpServerRequest.bodyHandler(buffer -> {
+                    this.topic = buffer.toJsonObject().getString("topic");
+                    this.partition = buffer.toJsonObject().getInteger("partition");
+                    if (buffer.toJsonObject().containsKey("offset")) {
+                        this.offset = Long.parseLong(buffer.toJsonObject().getString("offset"));
+                    }
+                    this.kafkaTopic = this.topic;
+                    this.setSubscribeHandler(subscribeResult -> {
+                        if (subscribeResult.succeeded()) {
+                            sendConsumerSubscriptionResponse(httpServerRequest.response());
+                        }
+                    });
+                    this.subscribe(false);
+                });
+
+                break;
+
+            case CONSUME:
+
+                this.setManualRecordBatchHandler(result -> {
+
+                    Buffer buffer = (Buffer) messageConverter.toMessages(result);
+
+                    sendConsumerRecordsResponse(httpServerRequest.response(), buffer);
+                });
+
+                if (httpServerRequest.getHeader("timeout") != null) {
+                    this.pollTimeOut = Long.parseLong(httpServerRequest.getHeader("timeout"));
+                }
+
+                this.consume();
```

Comment:
I think we don't need the `setManualRecordBatchHandler` method but we can pass the handler to the `consume` method directly. Or maybe I'm missing something?